webapp/views.py

```diff
@@ -1,139 +1,337 @@
+
+from django import template
+
 from webapp.services.services import services
 import logging
 
+register = template.Library()
 logger = logging.getLogger(__name__)
+SUPERUSER="stanislavgrozny@gmail.com"
 
-def home(request):
-    """Renders the home page."""
+def mylogin(request):
+    #cleanData()
+    #createSuperUser(request)
+    #print('check if signed in')
+    if(isUserSignedIntoGoogle(request)):
+       addIfNewUser(request)
+       url = reverse('home')
+       return HttpResponseRedirect(url)
+
+    """Renders the login page."""
+    #if user click login button
     assert isinstance(request, HttpRequest)
+    if request.method == "POST":
+       #print('Entered post')
+       return get_profile_required(request)
+       return HttpResponseRedirect('home')
+   #load login page
+    #print('not signed in')
     return render(
         request,
-        'webapp/index.html',
+        'webapp/login.html',
         {
-            'title':'Syndication Tool',
+            'title':'Login',
+            'message':'Your application description page.',
             'year':datetime.now().year,
-        }
-    )
+            
+        })
 
+@decorators.oauth_required
+def get_profile_required(request):
+    resp, content = request.oauth.http.request(
+        'https://www.googleapis.com/plus/v1/people/me')
+    return http.HttpResponse(content)
+
+@decorators.oauth_enabled
+def isUserSignedIntoGoogle(request):
+    if request.oauth.has_credentials():
+        return True
+    else: return False
+
+
+@decorators.oauth_enabled
+def addIfNewUser(request):
+    if request.oauth.has_credentials():
+       #print(request.oauth.credentials.id_token.items())
+       if User.objects.filter(username=request.oauth.credentials.id_token['email']).exists():
+           updateNameInfo=User.objects.get(username=request.oauth.credentials.id_token['email'])
+           #print('user exists') 
+		   ## Update the users name ( in the case that the user was created through group/role management pages)
+           updateNameInfo.set_first_name=request.oauth.credentials.id_token['given_name']
+           updateNameInfo.set_last_name=request.oauth.credentials.id_token['family_name']
+           updateNameInfo.save()         
+       else:
+	   #Create a new user from the logged in google info
+         newUser=User.objects.create_user(request.oauth.credentials.id_token['email'],request.oauth.credentials.id_token['email'],'tempPass')
+         newUser.first_name=request.oauth.credentials.id_token['given_name']
+         newUser.last_name=request.oauth.credentials.id_token['family_name']
+         newUser.save()
+       user = authenticate(username=request.oauth.credentials.id_token['email'], password='tempPass')
+       if user is not None:
+           #print('logging in user')
+           login(request, user)
+
+def logout_view(request):
+   logout(request)
+   #return HttpResponseRedirect('login')
+   return render(

```

Comment:
it looks like the token is being passed around and we can delete it ourselves:

http://stackoverflow.com/questions/30739352/django-rest-framework-token-authentication-logout

Can we do that here?